presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -863,7 +870,7 @@ public PlanWithProperties visitSemiJoin(SemiJoinNode node, PreferredProperties p
                     }
                     else {
                         source = withDerivedProperties(
-                                partitionedExchange(idAllocator.getNextId(), REMOTE, source.getNode(), createPartitioning(sourceSymbols), Optional.empty()),
+                                partitionedExchange(idAllocator.getNextId(), getExchangeScope(source.getNode()), source.getNode(), createPartitioning(sourceSymbols), Optional.empty()),
```

Comment:
This is in SemiJoin, do we still want to materialize exchange ?

Reply:
Yes, it might be beneficial for retriability.